### PR TITLE
feat(core): allow parameters to be hidden based on other params

### PR DESCRIPTION
**Pull request by anotherchrisberry**
A boring implementation of https://github.com/spinnaker/deck/pull/4942 with no web workers or eval

### CODE REVIEW

**Comment by jrsquared** [comment]
so boring.

**Comment by jrsquared** [comment]
💤

**Comment by jrsquared** [comment]
😴

**Comment by anotherchrisberry** [comment]
Wake me up when this pull request is over

File: app/scripts/modules/core/src/domain/IPipeline.ts [hunk 1]
```diff
@@ -28,6 +28,13 @@ export interface IParameter {
   'default': string;
   hasOptions: boolean;
   options: IParameterOption[];
+  comparator?: IParameterCondition;
```
Comment (jrsquared) [inline app/scripts/modules/core/src/domain/IPipeline.ts]: should this be `comparator?: IParameterCondition;`

File: app/scripts/modules/core/src/pipeline/manualExecution/manualPipelineExecution.controller.js [hunk 2]
```diff
@@ -129,14 +130,57 @@ module.exports = angular.module('spinnaker.core.pipeline.manualPipelineExecution
...
+      const { name } = parameterConfig;
+      const parameters = trigger ? trigger.parameters : {};
+      if (this.parameters[name] === undefined) {
+        this.parameters[name] = parameters[name] !== undefined ? parameters[name] : parameterConfig.default;
+      }
+    };
+
+    this.updateParameters = () => {
+      this.command.pipeline.parameterConfig.forEach(p => {
+        if (p.conditional) {
```
Comment (jrsquared) [inline app/scripts/modules/core/src/pipeline/manualExecution/manualPipelineExecution.controller.js]: although you call it conditional here.

File: app/scripts/modules/core/src/pipeline/manualExecution/manualPipelineExecution.controller.js [hunk 3]
```diff
@@ -129,14 +130,57 @@ module.exports = angular.module('spinnaker.core.pipeline.manualPipelineExecution
...
+            this.hiddenParameters.add(p.name);
+          } else {
+            this.hiddenParameters.delete(p.name);
+            this.addParameter(p);
+          }
+        }
+      });
+    };
+
+    this.shouldInclude = (p) => {
```
Comment (erikmunson) [inline app/scripts/modules/core/src/pipeline/manualExecution/manualPipelineExecution.controller.js]: ❤️

**Comment by erikmunson** [comment]
this makes me v bored, and v happy.

File: app/scripts/modules/core/src/domain/IPipeline.ts (hunk 1 above)
Comment (anotherchrisberry) [reply app/scripts/modules/core/src/domain/IPipeline.ts]: I don't know what you're talking about

**Comment by christopherthielen** [comment]
FWIW there isn’t he JavaScript SpEL evaluator used in timelines. 

I didn’t read this PR tho. Probably really good work.

**Pull request merged**